pyls/workspace.py

```diff
@@ -24,19 +23,28 @@ class Workspace(object):
     M_SHOW_MESSAGE = 'window/showMessage'
 
     def __init__(self, root_uri, lang_server=None):
-        self._url_parsed = urlparse(root_uri)
-        self.root = unquote(self._url_parsed.path)
+        self._root_uri = root_uri
+        self._root_uri_scheme = uris.urlparse(self._root_uri)[0]
+        self._root_path = uris.to_fs_path(self._root_uri)
         self._docs = {}
         self._lang_server = lang_server
 
+    @property
+    def root_path(self):
+        return self._root_path
+
+    @property
+    def root_uri(self):
+        return self._root_uri
+
     def is_local(self):
-        return (self._url_parsed.scheme == '' or self._url_parsed.scheme == 'file') and os.path.exists(self.root)
+        return (self._root_uri_scheme == '' or self._root_uri_scheme == 'file') and os.path.exists(self._root_path)
```

Comment:
I think it was used previously... we should add back support for non-local URIs and stop assuming things exist on disk. Let me create issues for these things.